openpype/hosts/maya/api/plugin.py

```diff
@@ -405,14 +405,21 @@ def collect_instances(self):
                 # No existing scene instance node for this layer. Note that
                 # this instance will not have the `instance_node` data yet
                 # until it's been saved/persisted at least once.
-                # TODO: Correctly define the subset name using templates
-                prefix = self.layer_instance_prefix or self.family
-                subset_name = "{}{}".format(prefix, layer.name())
+                project_name = self.create_context.get_current_project_name()
                 instance_data = {
-                    "asset": legacy_io.Session["AVALON_ASSET"],
-                    "task": legacy_io.Session["AVALON_TASK"],
+                    "asset": self.create_context.get_current_asset_name,
+                    "task": self.create_context.get_current_task_name,
```

Comment:
I assume these are functions like `get_current_project_name()` above?
```suggestion
                    "asset": self.create_context.get_current_asset_name(),
                    "task": self.create_context.get_current_task_name(),
```

Reply:
Fixed, wtf why it was working when I was testing it.

Reply:
Did you push the fix?

Reply:
It should have been pushed with https://github.com/ynput/OpenPype/pull/5442/commits/e6a847efb6425c266d6c7800b9a64a113bac2d18

I was pulling this branch from scratch and testing it right now and it works, please try again.